cmake/aws-c-cal-config.cmake

```diff
@@ -2,14 +2,21 @@ include(CMakeFindDependencyMacro)
 
 find_dependency(aws-c-common)
 
-if (NOT BYO_CRYPTO AND NOT WIN32 AND NOT APPLE)
-    list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/modules")
-    find_dependency(crypto)
-endif()
-
 if (BUILD_SHARED_LIBS)
     include(${CMAKE_CURRENT_LIST_DIR}/shared/@PROJECT_NAME@-targets.cmake)
 else()
     include(${CMAKE_CURRENT_LIST_DIR}/static/@PROJECT_NAME@-targets.cmake)
 endif()
 
+if (NOT BYO_CRYPTO AND NOT WIN32 AND NOT APPLE)
+    get_target_property(AWS_C_CAL_DEPS AWS::aws-c-cal INTERFACE_LINK_LIBRARIES)
+    # pre-cmake 3.3 IN_LIST search approach
+    list (FIND AWS_C_CAL_DEPS "OpenSSL::Crypto" _index)
```

Comment:
I mean..... by the time we're doing all this..... isn't it simpler to just remove all of the openssl and android branches and see if the problem goes away?

Reply:
So basically you suggest to completely drop the support of the build with the OpenSSL.

Reply:
I'm really sorry, but it is not "CPP SDK build failure", moreover, SDK itself builds perfectly fine.
It is "apps using aws-c-cal with OpenSSL crypto library form system build failure" because aws-c-cal declares a dependency on OpenSSL but never instructs it's consumers to find it during cmake configure.

Reply:
Comment added.